FlintCore/Core/Flint.swift

```diff
@@ -430,4 +431,8 @@ class PreconditionChangeObserver: PurchaseTrackerObserver, UserFeatureTogglesObs
     func userFeatureTogglesDidChange() {
         availabilityChecker.invalidate()
     }
+    
+    func permissionStatusDidChange(_ permission: SystemPermission) {
+        availabilityChecker.invalidate()
```

Comment:
Add a comment here that it's OK that we may be called on any thread by the permission adapter - the availability checker is threadsafe